### PR TITLE
Add CWE support in all importers

### PR DESCRIPTION
**Pull request by ziadhany**
issues: #1093

### CODE REVIEW

**Comment by TG1999** [comment]
@ziadhany Thanks++, before associating any vulnerability with CWE please check if it exists in cwe2 DB or not like this https://github.com/nexB/vulnerablecode/pull/1256, and also please do this for NVD importer too.

File: vulnerabilities/importers/github.py [hunk 1]
```diff
@@ -227,10 +233,18 @@ def process_response(resp: dict, package_type: str) -> Iterable[AdvisoryData]:
             else:
                 logger.error(f"Unknown identifier type {identifier_type!r} and value {value!r}")
 
+        weaknesses = []
```
Comment (TG1999) [inline vulnerabilities/importers/github.py:236]: Please make a separate function `get_cwes_from_github_advisory` and add docstring and tests for that.

File: vulnerabilities/importers/gitlab.py [hunk 2]
```diff
@@ -253,4 +271,102 @@ def parse_gitlab_advisory(file):
         references=references,
         date_published=date_published,
         affected_packages=affected_packages,
+        weaknesses=cwe_list,
     )
+
+
+class GitLabBasicImprover(Improver):
```
Comment (TG1999) [inline vulnerabilities/importers/gitlab.py:278]: Please remove this improver from here.

**Review by TG1999 — CHANGES_REQUESTED** [review]
@ziadhany thanks++, some nits for your consideration

File: vulnerabilities/importers/github.py (hunk 1 above)
Comment (ziadhany) [reply vulnerabilities/importers/github.py]: Done

File: vulnerabilities/importers/github.py [hunk 3]
```diff
@@ -227,10 +233,34 @@ def process_response(resp: dict, package_type: str) -> Iterable[AdvisoryData]:
...
+    db = Database()
+    cwe_list = get_item(advisory, "cwes", "nodes") or []
+    for cwe_item in cwe_list:
+        cwe_string = get_item(cwe_item, "cweId")
+        if cwe_string:
+            cwe_id = get_cwe_id(cwe_string)
+            try:
+                db.get(cwe_id)
+                weaknesses.append(cwe_id)
+            except Exception:
```
Comment (ziadhany) [inline vulnerabilities/importers/github.py:264]: we should change this general exception  ( InvalidCWEError ) after we merge this https://github.com/nexB/cwe2/pull/10

**Review by TG1999 — APPROVED** [review]
LGTM!

**Pull request merged**